### PR TITLE
8279877: Document IDEA IDE setup in docs/ide.md

### PR DESCRIPTION
**Pull request by djelinski**
Currently the information about using IDEA can only be found on mailing lists.

Additionally fixed the documentation for Visual Studio; the target `ide-project` does not exist.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8279877](https://bugs.openjdk.java.net/browse/JDK-8279877): Document IDEA IDE setup in docs/ide.md

### Reviewers
 * [Lance Andersen](https://openjdk.java.net/census#lancea) (@LanceAndersen - **Reviewer**)
 * [Erik Joelsson](https://openjdk.java.net/census#erikj) (@erikj79 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7044/head:pull/7044` \
`$ git checkout pull/7044`

Update a local copy of the PR: \
`$ git checkout pull/7044` \
`$ git pull https://git.openjdk.java.net/jdk pull/7044/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7044`

View PR using the GUI difftool: \
`$ git pr show -t 7044`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7044.diff">https://git.openjdk.java.net/jdk/pull/7044.diff</a>

</details>

### CODE REVIEW

File: doc/ide.md [hunk 1]
```diff
@@ -70,4 +75,24 @@ make compile-commands-hotspot
...
+
+The JDK project has a script that can be used for indexing the project
+with IntelliJ. After configuring and building the JDK, IntelliJ workspace
+can be generated using:
+
+```shell
+bash bin/idea.sh
+```
+
+This creates a subfolder called `.idea` in the current folder. To use it,
```
Comment (LanceAndersen) [inline doc/ide.md:88]: I might consider rewording this.  Typically this is run at the top of the your openjdk fork/clone.

**Review by LanceAndersen — COMMENTED** [review]
Thank you for taking a stab at this.  As far as I am aware, the ide.html should be generated from the ide.md so you should only need to modify ide.md

**Comment by djelinski** [comment]
> (...) As far as I am aware, the ide.html should be generated from the ide.md so you should only need to modify ide.md

Are you sure about this? The discussion on #5676 suggests that the HTML files need to be modified.

I wouldn't mind not having to edit htmls; `make update-build-docs` is broken on Cygwin, and I had to fix paths in compile commands manually.

**Comment by LanceAndersen** [comment]
> > (...) As far as I am aware, the ide.html should be generated from the ide.md so you should only need to modify ide.md
> 
> Are you sure about this? The discussion on #5676 suggests that the HTML files need to be modified.
> 
> I wouldn't mind not having to edit htmls; `make update-build-docs` is broken on Cygwin, and I had to fix paths in compile commands manually.

Yes pandoc as far as I am aware should be generating the html so lets wait for Erik to comment as it seems strange to have to do both and would be better to address the issue so only one file needs updated if there is indeed an issue.

**Comment by erikj79** [comment]
> > > (...) As far as I am aware, the ide.html should be generated from the ide.md so you should only need to modify ide.md
> > 
> > 
> > Are you sure about this? The discussion on #5676 suggests that the HTML files need to be modified.
> > I wouldn't mind not having to edit htmls; `make update-build-docs` is broken on Cygwin, and I had to fix paths in compile commands manually.
> 
> Yes pandoc as far as I am aware should be generating the html so lets wait for Erik to comment as it seems strange to have to do both and would be better to address the issue so only one file needs updated if there is indeed an issue.

You should only manually edit the .md files and then let pandoc generate .html files using `make update-build-docs`. Both types of files should then be included in the commit. If you have trouble generating .html files this way, you will need help from someone who can. Let me know when the textual content of this patch is considered final, then I can do it for you if you like.

File: doc/ide.md [hunk 2]
```diff
@@ -70,4 +75,24 @@ make compile-commands-hotspot
 
 ### IDE support for Java code
 
-This section is a work in progress.
\ No newline at end of file
+#### IntelliJ IDEA
+
+The JDK project has a script that can be used for indexing the project
+with IntelliJ. After configuring and building the JDK, IntelliJ workspace
```
Comment (erikj79) [inline doc/ide.md:81]:
```suggestion
with IntelliJ. After configuring and building the JDK, an IntelliJ workspace
```

File: doc/ide.md [hunk 3]
```diff
@@ -70,4 +75,24 @@ make compile-commands-hotspot
...
+can be generated using:
+
+```shell
+bash bin/idea.sh
+```
+
+This creates a subfolder called `.idea` in the current folder. To use it,
+choose `File -> Open...` in IntelliJ and select the folder where you run
```
Comment (erikj79) [inline doc/ide.md:89]:
```suggestion
choose `File -> Open...` in IntelliJ and select the folder where you ran
```

File: doc/ide.md [hunk 4]
```diff
@@ -45,12 +45,17 @@ after the workspace has been generated.
 
 #### Visual Studio
 
-This section is a work in progress.
+The make system can generate a Visual Studio project for a limited subset
+of native sources. After configuring, the project can be generated using:
```
Comment (erikj79) [inline doc/ide.md:49]: The limited subset is hotspot, which is a pretty clear separation. 
```suggestion
The make system can generate a Visual Studio project for the Hotspot 
native source. After configuring, the project is be generated using:
```

**Review by erikj79 — COMMENTED** [review]
Added some suggestions on language.

**Comment by djelinski** [comment]
Thanks for all your suggestions. Let me know if the current version is acceptable.

Re pandoc, it looks like the Windows version is generating similar output to the Linux one, so I think I can handle this. Thanks for the offer.

**Review by erikj79 — APPROVED** [review]
Contents of ide.md looks good. As long as ide.html is generated correctly using pandoc I'm good with this.

**Review by LanceAndersen — APPROVED** [review]
I think you are good to go with your latest set of changes.

Not sure if you want to refer to the idea.sh -h for additional options (though that does not seem to give you any detail regarding the options)

**Comment by djelinski** [comment]
> Not sure if you want to refer to the idea.sh -h for additional options

I'm not sure about that either. These options are usually not required, and the -h output leaves something to be desired. I'd rather keep the current version, and only expand if users start asking.

/integrate

**Comment by LanceAndersen** [comment]
> I think you are good to go with your latest set of changes.
> 
> Not sure if you want to refer to the idea.sh -h for additional options (though that does not seem to give you any detail regarding the options)



> > Not sure if you want to refer to the idea.sh -h for additional options
> 
> I'm not sure about that either. These options are usually not required, and the -h output leaves something to be desired. I'd rather keep the current version, and only expand if users start asking.
> 
> /integrate

Works for me.

BTW typically we do the /integrate in a separate comment but it seems to have worked her

**Pull request closed without merging**